File: Data Structures & Algorithms/binary-search/submission-5.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size();
        int index;
        
        while(l < r){
            index =  (l+r) / 2;
            if(nums[index] == target)
                return index;
            else if(nums[index] > target)
                r = index;
            else if (nums[index] < target){
                if(index == l)
                    l++;
                else
                    l = index;
            }
        }

        index =  (l+r) / 2;
        if(nums[index] == target)
                return index;
            

        return -1;
    }
};

File: Data Structures & Algorithms/contains-duplicate-ii/submission-5.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        windowMap: dict = defaultdict(int)
        l = 0
        for r in range(len(nums)):
            width = abs(r-l)
            if width > k:
                windowMap[nums[l]] -= 1
                l += 1

            if windowMap[nums[r]] >= 1:
                return True

            windowMap[nums[r]] += 1

        return False